mlir/lib/Dialect/EmitC/IR/EmitC.cpp

```diff
@@ -68,6 +68,8 @@ static LogicalResult verifyInitializationAttribute(Operation *op,
            << "string attributes are not supported, use #emitc.opaque instead";
 
   Type resultType = op->getResult(0).getType();
+  if (auto lType = dyn_cast<LValueType>(resultType))
```

Comment:
This also allows `ConstantOp` to have LValue type, right? I think it should be restricted to VariableOps

Reply:
Correct, I haven't gone through all op definitions yet to restrict operands/results to (non)lvalue types yet. The current state is more or less that variable ops return lvalues and all tests pass again.

Reply:
This should be handled now as the lvalue type is disallowed in the "EmitCType" predicate.